tests/contrib/test_gce.py

```diff
@@ -14,102 +14,195 @@
+import tempfile
 
 __author__ = 'jcgregorio@google.com (Joe Gregorio)'
 
+METADATA_SERVER = 'oauth2client.contrib.gce._get_metadata'
+DEFAULT_SERVICE_ACCOUNT = {'email': 'default'}
+A_SERVICE_ACCOUNT = {
+    'scopes': [
+        'https://www.googleapis.com/auth/cloud-platform',
+        'https://www.googleapis.com/auth/iam'
+    ],
+    'email': '12345678-compute@developer.gserviceaccount.com'
+}
+
+GET_ACCESS_TOKEN = 'oauth2client.contrib.gce._get_access_token'
+ONE_SECOND_TOKEN = {'access_token': '12345abcde', 'expires_in': 1}
+FOREVER_TOKEN = {'access_token': '12345abcde', 'expires_in': 999999999}
+
 
 class AppAssertionCredentialsTests(unittest2.TestCase):
 
+    # BASIC TESTS
+
     def test_constructor(self):
         credentials = AppAssertionCredentials(foo='bar')
-        self.assertEqual(credentials.scope, '')
         self.assertEqual(credentials.kwargs, {'foo': 'bar'})
         self.assertEqual(credentials.assertion_type, None)
 
     @mock.patch('warnings.warn')
-    def test_constructor_with_scopes(self, warn_mock):
-        scope = 'http://example.com/a http://example.com/b'
-        scopes = scope.split()
-        credentials = AppAssertionCredentials(scope=scopes, foo='bar')
-        self.assertEqual(credentials.scope, scope)
-        self.assertEqual(credentials.kwargs, {'foo': 'bar'})
-        self.assertEqual(credentials.assertion_type, None)
+    @mock.patch(METADATA_SERVER, return_value=A_SERVICE_ACCOUNT)
+    def test_constructor_with_matching_scopes(self, get_metadata, warn_mock):
+        scope = 'https://www.googleapis.com/auth/cloud-platform'
+        AppAssertionCredentials(scope=scope)
         warn_mock.assert_called_once_with(_SCOPES_WARNING)
 
-    def test_to_json_and_from_json(self):
-        credentials = AppAssertionCredentials()
-        json = credentials.to_json()
-        credentials_from_json = Credentials.new_from_json(json)
-        self.assertEqual(credentials.access_token,
-                         credentials_from_json.access_token)
-
-    def _refresh_success_helper(self, bytes_response=False):
-        access_token = u'this-is-a-token'
-        return_val = json.dumps({u'access_token': access_token})
-        if bytes_response:
-            return_val = _to_bytes(return_val)
-        http = mock.MagicMock()
-        http.request = mock.MagicMock(
-            return_value=(mock.Mock(status=http_client.OK), return_val))
+    @mock.patch(METADATA_SERVER, return_value=A_SERVICE_ACCOUNT)
+    def test_construtor_with_missing_scopes(self, get_metadata):
+        scope = 'http://wwww.nobodyhasthisscope.com'
+        error = None
+        try:
+            AppAssertionCredentials(scope=scope)
+        except ValueError as e:
+            error = e
+        self.assertIsNotNone(error)
 
+    @mock.patch(METADATA_SERVER, return_value=A_SERVICE_ACCOUNT)
+    def test_default_service_account_info(self, get_metadata):
+        credentials = AppAssertionCredentials()
+        self.assertEquals(
```

Comment:
Just noticed the `assertEquals` instead of `assertEqual`. The plural is moving towards deprecation, so prefer `assertEqual`:

> The plurals are undocumented. Keep them that way to discourage use.